### ISSUE TITLE
Brainstorming: Passing parameter via cross application navigation

### ISSUE THREAD

**Issue by axelmohnen** [abap2UI5/ext-fiori_launchpad_on_premise#5]
Hi @oblomov-dev,

I have the requirement to pass data back from called ABAP2UI5 application to the calling Fiori element app.
Within normal SAPUI5 apps you could archive this via cross application navigation using some events from the extension API.
e.g "modifyStartupExtension", "adaptNavigationContext", "doCustomNavigation", etc...

Schema:
FLP -> Calling app (Fiori element) -> Intent based Navigation -> Called app (ABAP2UI5)
           Calling app (Fiori element) <- Intent based Navigation (with returning data) <- Called app (ABAP2UI5)

In ABAP2UI5 we have to return the data via variable "sap.z2ui5" I guess (like for APP_START) just the other way around and inside the "View1.controller.js" we have to push the data to the navigation context somehow.

Did you every had such a strange requirement ;-)
Do you have an idea how to implement this?
Thanks a lot.
Kind regards
Axel

**Comment by oblomov-dev** [comment]
Hi @Axel,

never used cross application navigation, but sounds like an interesting topic:

That means as a first step you want to call a normal fiori app out of abap2ui5, so abap2ui5 is the caller and we have to push data to the navigation context?

Maybe adding a frontend function is the right way for this. We can extend abap2ui5 at this position:
https://github.com/abap2UI5/abap2UI5/blob/7a3f9e845ca86975aece6ae3c66b9183d7e6b075/src/z2ui5_cl_fw_http_handler.clas.abap#L244C84-L244C84

I also used this position to call external URLs, can you share the exact JS calls you want to execute that we can add it there?

Best regards.

**Comment by axelmohnen** [comment]
No, it's the opposite. Fiori app is the caller via target mapping. The abap2ui5 app is the called app.
The abap2ui5 should return specific parameter back to the calling Fiori app.

I will try to build up an example in order to see the exact JS calls.
